src/Feldspar/Compiler/Imperative/Representation.hs

```diff
@@ -90,8 +90,7 @@ data Entity t
         , loopBody                  :: Bool
         -- Is this a loopbody in disguise?
         , inParams                  :: [Variable t]
-        -- Left is regular return, right is fast return.
-        , outParams                 :: Either [Variable t] (Variable t)
+        , procType                  :: Type
```

Comment:
This change limits the number of return values to one. That might be OK but it's not obvious to me that it is sufficient for Wool/OpenMP loop bodies.

Reply:
As for Wool loop bodies, they have C semantics. If you want to change a free variable, you have to pass a pointer to it to the loop body function. OpenMP loop bodies are in line and ned not be treated specially.

As for the main program, we still use a singel "out" parameter to pass the result (unless we set fastReturn). FromCore explicitly adds a level of indirection to "out"; if the type of the "return value" is T then out will have the type '1 :# Pointer T'. If we want to return multiple results, we can still do that using "out0", "out1", ... or some similar pattern.

Reply:
My understanding is that this change is supposed to produce the same output which would be nice to verify.

I had a quick look and I can't find any gold tests for OpenMP/Wool, not sure if I was thinking at the time I added the options. Could you make a separate pull request adding a gold test for OpenMP and a gold test for Wool? The Wool configuration is called sicsOptions3 and the OpenMP configuration is called c99OpenMpPlatformOptions.